test/uri.jl

```diff
@@ -477,4 +478,27 @@ end
     # Issue 323
     @test string(HTTP.URI(scheme="http", host="example.com")) == "http://example.com"
 
+    @testset "Normalize URI paths" begin
+        # Examples given in https://tools.ietf.org/html/rfc3986#section-5.2.4
+        @test URIs.normpath("/a/b/c/./../../g") == "/a/g"
+        @test URIs.normpath("mid/content=5/../6") == "mid/6"
+
+        checknp = (x, y)->(@test URIs.normpath(URI(x)) == URI(y))
+
+        # "Abnormal" examples in https://tools.ietf.org/html/rfc3986#section-5.4.2
+        checknp("http://a/b/c/d/../../../g", "http://a/g")
+        checknp("http://a/b/c/d/../../../../g", "http://a/g")
+
+        # "Normal" examples
+        checknp("http://a", "http://a")
+        checknp("http://a/b/c/.", "http://a/b/c/")
+        checknp("http://a/b/c/./", "http://a/b/c/")
+        checknp("http://a/b/c/..", "http://a/b/")
+        checknp("http://a/b/c/../", "http://a/b/")
+        checknp("http://a/b/c/../g", "http://a/b/g")
+        checknp("http://a/b/c/../..", "http://a/")
+        checknp("http://a/b/c/../../", "http://a/")
+        checknp("http://a/b/c/../../g", "http://a/g")
+    end
```

Comment:
If anyone, like me, was wondering, difference from `Base.normpath`
according to these tests are:

`Base.normpath` tranlates `http://` into `http:/`.

`Base.normpath` translates `http://a/b/c/d/../../../../g` into , "http:/g")

Reply:
The difference is because `http://a` is not considered part of the path; the path begins with the first `/` after `a`.